daemon/container_unix.go

```diff
@@ -976,9 +987,9 @@ func (container *Container) connectToNetwork(idOrName string, updateSettings boo
 		if err != nil {
 			return err
 		}
-	}
 
-	container.updateSandboxNetworkSettings(sb)
+		container.updateSandboxNetworkSettings(sb)
```

Comment:
Because it is supposed to be called only once, when the sandbox is created. Not every time the sandbox is retrieved from the sandbox database. You can call it multiple times but there really is no need. Please let me know if you need more clarification.